extension/parquet/include/row_number_column_reader.hpp

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// generated_column_reader.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef DUCKDB_AMALGAMATION
+#include "duckdb/common/limits.hpp"
+#endif
+#include "column_reader.hpp"
+#include "templated_column_reader.hpp"
+
+namespace duckdb {
+
+//! Reads a constant that is not read from the parquet file but instead generated by the parquet reader, used for
+//! filename column. Can be used for handling hive partitioning later.
+class RowNumberColumnReader : public ColumnReader {
+public:
+	RowNumberColumnReader(ParquetReader &reader, LogicalType type_p, const SchemaElement &schema_p, idx_t schema_idx_p,
+	                      idx_t max_define_p, idx_t max_repeat_p);
+
+public:
+	idx_t Read(uint64_t num_values, parquet_filter_t &filter, uint8_t *define_out, uint8_t *repeat_out,
+	           Vector &result) override;
+
+	unique_ptr<BaseStatistics> Stats(const std::vector<ColumnChunk> &columns) override {
```

Comment:
Can we maybe add the stats here? this allows proper handling of pushed down filters on this column